signer.js

```diff
@@ -1,32 +1,91 @@
 'use strict'
 
-const cookieSignature = require('cookie-signature')
+const crypto = require('crypto')
 
-module.exports = function (secret) {
+function signerFactory (secret, factoryAlgorithm = 'sha256') {
   const secrets = Array.isArray(secret) ? secret : [secret]
-  const [signingKey] = secrets
+
+  for (const secret of secrets) {
+    if (typeof secret !== 'string') {
+      throw new TypeError('Secret key must be a string.')
+    }
+  }
+
+  const signingKey = secrets[0]
+
+  // validate that the algorithm is supported by the node runtime
+  try {
+    crypto.createHmac(factoryAlgorithm, signingKey)
+  } catch (e) {
+    throw new TypeError(`Algorithm ${factoryAlgorithm} not supported.`)
+  }
 
   return {
-    sign (value) {
-      return cookieSignature.sign(value, signingKey)
+    sign (value, secret = signingKey, algorithm = factoryAlgorithm) {
+      if (typeof value !== 'string') {
+        throw new TypeError('Cookie value must be provided as a string.')
+      }
+      return value + '.' + crypto
+        .createHmac(algorithm, secret)
+        .update(value)
+        .digest('base64')
+        // remove base64 padding (=) as it has special meaning in cookies
+        .replace(/=+$/, '')
```

Comment:
Please extract the cookie-signature bits into its own file and include the relative license disclaimer in there.

Reply:
I would rather not extract the cookie-signature bits in its own file, as it is imho now a "logical unit"